tests/evm_test.go

```diff
@@ -58,6 +61,16 @@ func testVMCase(t *testing.T, name string, c *VMCase) {
 	}
 
 	e, _ := executor.BeginTxn(root, c.Env.ToHeader(t))
+	{
+		// todo: workaround fix to pass the test
+		// we need to consider best way to give GasPrice and Origin as context value into evm in test
+		t := reflect.ValueOf(e).Elem()
+		ctx := t.FieldByName("ctx")
+		gasPrice := (*types.Hash)(unsafe.Pointer(ctx.FieldByName("GasPrice").UnsafeAddr()))
+		origin := (*types.Address)(unsafe.Pointer(ctx.FieldByName("Origin").UnsafeAddr()))
+		*gasPrice = types.BytesToHash(env.GasPrice.Bytes())
+		*origin = env.Origin
+	}
```

Comment:
I just noticed that the `Context()` method of the `Transition` struct isn't used anywhere, so this being the only place using it, you could just change the signature of it to return the pointer instead of the value and then do:
```
e.Context().GasPrice = types.BytesToHash(env.GasPrice.Bytes())
e.Context().Origin = env.Origin
```

Again, this might not be the proper solution either - I would love to hear an opinion from @lazartravica @zivkovicmilos

Reply:
@Vuksan Yes, you're correct. I checked it on my side. Transition.Context() might not be called from anywhere. And if I change this function to return pointer instead of value, we can pass the test without using reflect. But strange thing is that Transition has Context() and GetTxContext(). Both method return same value

Reply:
Yep, this is probably just an oversight. We can convert the Transition.Context() method to return a pointer, probably change the name as well, to something like Transition.GetContextPtr() - as @Vuksan suggested.

Reply:
I'd like to get feedback from @lazartravica, if it's okay.

Reply:
> Yep, this is probably just an oversight. We can convert the Transition.Context() method to return a pointer, probably change the name as well, to something like Transition.GetContextPtr() 

I agree with this 👍